tools/scripts/update-docker-tags.sh

```diff
@@ -18,12 +18,16 @@ export OLD_TAG=$(cat tools/scripts/environments-target.txt)
 export NEW_TAG="$1"
 
 # get list of images to replace via OLD_TAG in bumpenvs.yaml
-export IMAGES=$(grep -oP "(?<=new: ).*(?=,)" tools/scripts/bumpenvs.yaml | grep -F $OLD_TAG)
+export IMAGES=$(grep -oP "(?<=new: ).*(?=,)" tools/scripts/bumpenvs.yaml | grep -F :$OLD_TAG)
 
 # update tags on dockerhub
 for NAME in $IMAGES; do
     if [ "$2" == "--release" ]; then
-        export NEW_NAME="$(echo $NAME | grep -oP '.*(?=-dev:)'):$NEW_TAG"
+        if [ -z $(echo $NAME | grep -oP '.*(?=-dev:)') ]; then
+            export NEW_NAME="$(echo $NAME | grep -o '.*:')$NEW_TAG"
+        else
+            export NEW_NAME="$(echo $NAME | grep -oP '.*(?=-dev:)'):$NEW_TAG"
+        fi
```

Comment:
Actually, looking at this...

All that needs to happen is replacing `-dev:` with `:` and append new tag after new name, right?
```suggestion
        # replace everything after first colon with $NEW_TAG
        NEW_NAME=${NAME%%:*}:$NEW_TAG
        # remove -dev from new name if present
        NEW_NAME=${NEW_NAME/-dev:/:/}
        export NEW_NAME
```

I don't know if that's easier to read or not.  I think it's easier, but you tell me.

Reply:
There is one exception (for a set of images that we want to deprecate anyways) where `determinedai/deepspeed-cuda-gpt-neox` has no `-dev` repo.